sdks/python/apache_beam/dataframe/convert.py

```diff
@@ -67,7 +69,28 @@ def to_dataframe(
       expressions.PlaceholderExpression(proxy, pcoll))
 
 
+# PCollections generated by to_pcollection are memoized.
+# WeakValueDictionary is used so the caches are cleaned up with the parent
+# pipelines
```

Comment:
Nice. Perhaps it's worth noting that the pipeline (indirectly) holds references to the transforms which keep both the collections and expressions alive. (Keeping the expressions alive is important to ensure their ids never get accidentally re-used.)

Reply:
Good idea, added that language